CleanDiffuserTeam/CleanDiffuser
# Add pip installation support and package versioning

This PR adds support for pip installation and introduces proper package versioning for CleanDiffuser.

Changes:
1. Added `pyproject.toml` with project metadata and build system requirements
2. Added `setup.py` for backwards compatibility with older pip versions
3. Updated `cleandiffuser/__init__.py` to include a `__version__` attribute

Motivation:
- Enable easy installation via pip, improving accessibility and ease of use
- Introduce proper versioning to track package releases and ensure version consistency

Implementation details:
- The package version is now defined in `cleandiffuser/__init__.py` and is referenced by both `pyproject.toml` and `setup.py`
- This allows for a single source of truth for the package version, making it easier to manage and update

Testing:
- Verified successful installation using `pip install .` in a clean virtual environment
- Confirmed the package is importable after installation
- Checked that the version can be accessed via `cleandiffuser.__version__`

How to test this PR:
You can install the package directly from this PR using pip:
```sh
pip install git+https://github.com/imataka7/CleanDiffuser
```
This will allow you to test the installation process and verify that the package works as expected.

Please review and let me know if any adjustments are needed.
